src/libraries/System.Formats.Tar/tests/TarWriter/TarWriter.WriteEntryAsync.File.Tests.Unix.cs

```diff
@@ -149,5 +149,40 @@ await using (TarReader reader = new TarReader(archive))
                 }
             }, format.ToString(), new RemoteInvokeOptions { RunAsSudo = true }).Dispose();
         }
+        
+        [ConditionalFact(nameof(IsRemoteExecutorSupportedAndPrivilegedProcess))]
+        public void CreateEntryFromFileOwnedByNonExistentGroup_Async()
+        {
+            RemoteExecutor.Invoke(async () =>
+            {
+                string groupName = Path.GetRandomFileName()[0..6];
+                int groupId = CreateGroup(groupName);
+
+                using TempDirectory root = new TempDirectory();
+
+                string fileName = "file.txt";
+                string filePath = Path.Join(root.Path, fileName);
+                File.Create(filePath).Dispose();
+
+                SetGroupAsOwnerOfFile(groupName, filePath);
+
+                DeleteGroup(groupName);
+
+                await using MemoryStream archive = new MemoryStream();
+                await using (TarWriter writer = new TarWriter(archive, TarEntryFormat.Ustar, leaveOpen: true))
```

Comment:
Is this worth testing with all formats, not just ustar?

Reply:
Yeah, we can do that. It only applies to PosixTarEntry instances though, so V7 is excluded (no GName field).